src/ui/win32/Dlg_MessageBox.cpp

```diff
@@ -0,0 +1,121 @@
+#include "Dlg_MessageBox.hh"
+
+#include "RA_Defs.h" // for NativeStr
+
+#include "ui/viewmodels/MessageBoxViewModel.hh"
+
+#include <WTypes.h>
+
+#undef GetMessage // ignore Windows macro
+
+namespace ra {
+namespace ui {
+namespace win32 {
+
+using ra::ui::viewmodels::MessageBoxIcon;
+using ra::ui::viewmodels::MessageBoxButtons;
+using ra::ui::viewmodels::MessageBoxViewModel;
+
+static HRESULT(WINAPI *pTaskDialog)(HWND hwndParent, HINSTANCE hInstance, PCWSTR pszWindowTitle, PCWSTR pszMainInstruction,
+    PCWSTR pszContent, TASKDIALOG_COMMON_BUTTON_FLAGS dwCommonButtons, PCWSTR pszIcon, int *pnButton) = nullptr;
+
+Dlg_MessageBox::Controller::Controller() noexcept
+{
+    // TaskDialog isn't supported on WinXP, so we have to dynamically find it.
+    auto hDll = LoadLibraryA("comctl32");
+    if (hDll)
+        pTaskDialog = (HRESULT(WINAPI*)(HWND, HINSTANCE, PCWSTR, PCWSTR, PCWSTR, TASKDIALOG_COMMON_BUTTON_FLAGS, PCWSTR, int *))GetProcAddress(hDll, "TaskDialog");
+}
```

Comment:
TaskDialog doesn't exist before Vista. 
This function will probably fail on an XP device (which would be undefined by the standard). You should probably just get rid of this. and surround the whole class with a check to see if the platform is Vista or higher.

Reply:
I _did_ test this in XP. That's the whole reason I'm using `GetProcAddress` instead of the #define already present in the Windows.h. `GetProcAddress` returns NULL on XP, so it'll go into the non-TaskDialog section of the code.

As mentioned above, we can't conditionally compile anything out of the library because the exact same library has to work on everything from XP to 10.

You're right - `FreeLibrary` _should_ be getting called, but as long as we're expecting to use `pTaskDialog`, we can't.  I guess I could make hDll a member variable and call FreeLibrary in the destructor.

Reply:
Sounds good to me.